### ISSUE TITLE
docker.errors.TLSParameterError

### ISSUE THREAD

**Issue by sryze** [docker/for-win#2706]
Every time I run docker-compose I get the same error (see below). It worked fine yesterday, I haven't changed anything.

I'm using the Hyper-V version of Docker. Hyper-V is enabled.

Docker Toolbox works fine with my compose file. Also on Linux docker-compose works too.

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 16A5EF7D-D107-454F-99A9-37FAD45BA40F/20181006230827

### Expected behavior

Start services in docker-compose config.

### Actual behavior

```
docker-compose up
Traceback (most recent call last):
  File "docker-compose", line 6, in <module>
  File "compose\cli\main.py", line 71, in main
  File "compose\cli\main.py", line 124, in perform_command
  File "compose\cli\command.py", line 38, in project_from_options
  File "compose\cli\docker_client.py", line 84, in tls_config_from_options
  File "site-packages\docker\tls.py", line 81, in __init__
docker.errors.TLSParameterError: Path to a certificate and key files must be provided through the client_config param. TLS configurations should map the Docker CLI client configurations. See https://docs.docker.com/engine/articles/https/ for API details.
[1732] Failed to execute script docker-compose
```

### Information

  - Windows Version: 10
  - Docker for Windows Version: 18.06.1-ce-win73 (19507)

### Steps to reproduce the behavior

  1. Start Docker
  2. Run docker-compose up

**Comment by butaminas** [comment]
Same here. Tried to use Docker Toolbox only at first but then decided to switch to Docker and uninstall Docker Toolbox and now I get the same error when trying to do something with docker.

**Comment by GHDavid** [comment]
I have the same error. Like #2 I switched between Docker Toolbox and Docker for Windows versions. Is there a solution for our problem?

**Comment by GHDavid** [comment]
Finally I solved my problem with the reset my PC of windows 10 option. I selected the option of keep my personal files and all works fine with Docker.

**Comment by sryze** [comment]
I was able to use Docker again after I removed all the environment variables that were (I think) set by Docker Toolbox (`DOCKER_*`).

**Comment by butaminas** [comment]
> I was able to use Docker again after I removed all the environment variables that were (I think) set by Docker Toolbox (`DOCKER_*`).

Where did you removed those environment variables?

**Comment by sryze** [comment]
System properties -> Advanced system settings -> Environment variables -> User variables

**Comment by MrCooper42** [comment]
I am still having this issue.

**Comment by maciej-pyszynski** [comment]
@MrCooper42 After removing DOCKER_ entries from environment variables (Advanced system settings -> Environment variables -> User variables), restart docker/computer

**Comment by kbeck98** [comment]
@sryze thanks for posting that solution, worked perfectly for me

**Comment by fabienheureux** [comment]
My two cents: in my case, on a newly installed computer, it was because `docker login` had not been run 😃

**Comment by danielszuk** [comment]
> I was able to use Docker again after I removed all the environment variables that were (I think) set by Docker Toolbox (`DOCKER_*`).

Works like a charm, one addition: you have to restart machine, after remove variables

**Comment by octogod** [comment]
`eval $(docker-machine env -u)
`

did the trick for me

**Comment by purescripts** [comment]
Removed DOCKER_* from environment variables
Ran eval $(docker-machine env -u).

Good to go.

**Comment by gsusI** [comment]
To me, this was related with FILE SHARING options, allowing Docker to access to the drive from Docker Desktop settings pannel solved my issue.

I was using Docker Desktop 2.1.14 with WSL 2 Tech Preview enabled.

**Comment by rintoug** [comment]
> DOCKER_

This solved my problem too. Thanks!

**Comment by joniba** [comment]
none of the above solutions are working for me... anyone else have a solution to this?

**Comment by russmack** [comment]
Check if `echo $DOCKER_CERT_PATH` returns a value.
If so, try setting it to empty `DOCKER_CERT_PATH=`.
Then run your `docker-compose ...` command.

**Comment by nikhil-allyio** [comment]
Still facing this issue . Don't have any environment variables associated to DOCKER . Ran docker login as well -- no luck so far!

**Comment by baosen** [comment]
For WSL2, I was able to fix this issue by installing the latest 1.26.x binary found here: https://github.com/docker/compose/releases in /usr/local/bin/, then set DOCKER_HOST=unix:///var/run/docker.sock before running docker-compose. Don't use version 1.25.x.
